Library/Environment.swift

```diff
@@ -49,6 +49,9 @@ public struct Environment {
   /// A delegate to handle Facebook initialization and incoming url requests
   public let facebookAppDelegate: FacebookAppDelegateProtocol
 
+  /// A function that returns whether iOS 12 is available
+  public let isIOS12Available: () -> Bool
```

Comment:
Tried that..didn't work

Reply:
I'm not entirely sure what do you mean by `assertFailure`. We kinda need the `Environment` struct to store the value locally during the whole lifecycle of the app/tests so there has to be an underlying property / variable that we store this in (not sure how to achieve this with dynamic values - unless we use a key-value store / dictionary). Also, since we only support 3 major OS versions (`iOS 12`, `iOS 11`, `iOS 10`) it's not that much properties we'd eventually have to add to `Environment` so I think it should be fine.

For now, I have removed the ability to use `Environment` for `#availabile` checks in order for us to being able to merge this. We can iterate in a follow up PR.